kotlinx-coroutines-core/jvm/src/scheduling/WorkQueue.kt

```diff
@@ -138,23 +155,41 @@ internal class WorkQueue {
             } // Failed -> someone else stole it
         }
 
+        return pollWithMode(onlyBlocking = true /* only blocking */)
+    }
+
+    fun pollCpu(): Task? {
+        while (true) { // Poll the slot
+            val lastScheduled = lastScheduledTask.value ?: break
+            if (lastScheduled.isBlocking) break
+            if (lastScheduledTask.compareAndSet(lastScheduled, null)) {
+                return lastScheduled
+            } // Failed -> someone else stole it
+        }
```

Comment:
Why not move this code to `pollWithMode` as well? Only one line would need to change to
```kotlin
            if (lastScheduled.isBlocking == onlyBlocking) break
```

Reply:
Good suggestion, applied